include/seqan3/alignment/configuration/align_config_aligned_ends.hpp

```diff
@@ -0,0 +1,600 @@
+
+    /*!\name Constructor, destructor and assignment
+     * \brief Defaulted all standard constructor.
+     * \{
+     */
+    constexpr seq_end_gap_base()                                     noexcept = default;
+    constexpr seq_end_gap_base(seq_end_gap_base const &)             noexcept = default;
+    constexpr seq_end_gap_base(seq_end_gap_base &&)                  noexcept = default;
+    constexpr seq_end_gap_base & operator=(seq_end_gap_base const &) noexcept = default;
+    constexpr seq_end_gap_base & operator=(seq_end_gap_base &&)      noexcept = default;
+    ~seq_end_gap_base()                                              noexcept = default;
+
+    /*!\brief Constructs the wrapped value from the passed object.
+     * \tparam    _value_t Either a std::integral_constant or bool.
+     * \param[in] v        The value to wrap.
+     */
+    template <typename _value_t>
+    explicit constexpr seq_end_gap_base(_value_t const v) noexcept
+    {
+        if constexpr (std::is_same_v<_value_t, bool>)
+            value = v;
+        else
+            value = v();
+    }
+    //!}
+
+public:
+
+    //!\brief Returns the wrapped value.
+    constexpr bool operator()() const noexcept
+    {
+        return value;
+    }
+
+private:
+
+    //!\brief The wrapped value.
+    bool value{static_value};
+};
+
+// ----------------------------------------------------------------------------
+// seq1_leading
+// ----------------------------------------------------------------------------
+
+/*!\brief The penalty configuration for a leading gap in the first sequence of a pairwise sequence alignment.
+ * \ingroup configuration
+ * \tparam value_t The type of the value to be wrapped. Can be of type std::true_type, std::false_type or bool.
+ *
+ * \details
+ *
+ * This configuration element enables/disables penalties for the respective gaps in the pairwise
+ * sequence alignment. If the wrapped value evaluates to true, leading gaps in the first sequence will be not penalized.
+ * If one constructs this element with a std::integral_constant it will convert to a static type such that
+ * compile time optimizations can be enabled. If the type is constructed from a bool it will convert to a dynamic type
+ * but will be later during the instantiation of the pairwise alignment converted to a static type.
+ * Using a `bool` allows to dynamically set the value if the option is only known at runtime. If the option is
+ * already known at compile time the static version will be a better option.
+ *
+ * \see seq1_trailing
+ * \see seq2_leading
+ * \see seq2_trailing
+ */
+template <typename value_t>
+class seq1_leading : public seq_end_gap_base<value_t>
+{
+    //!\brief An alias for the base type.
+    using base_t = seq_end_gap_base<value_t>;
+
+public:
+
+    //!\privatesection
+    //!\brief An internal id to allow consistency checks with other gap specifiers.
+    static constexpr std::integral_constant<uint8_t, 0> id{};
```

Comment:
No they are not API. That's why they are documented in a privatesection. They are for internal use only, but I want to avoid making them private and befriendnig with the classes that access them.